app/services/set_localization_service.rb

```diff
@@ -0,0 +1,66 @@
+class SetLocalizationService
+  attr_reader :user, :http_accept_header
+  include Redmine::I18n
+
+  def initialize(user, http_accept_header)
+    @user = user
+    @http_accept_header = http_accept_header
+  end
+
+  ##
+  # Sets the locale.
+  # The locale is determined with the following priority:
+  #
+  #   1. The language as configured by the user.
+  #   2. The first language defined in the Accept-Language header sent by the browser.
+  #   3. OpenProject's default language defined in the settings.
+
+  def perform
+    lang = user_language || header_language || default_language
+
+    set_language_if_valid(lang)
+  end
+
+  private
+
+  def user_language
+    find_language(user.language) if user.logged?
+  end
+
+  def header_language
+    accept_lang = parse_qvalues(http_accept_header).first
+
+    lang = if accept_lang
+             accept_lang = accept_lang.downcase
+             find_language(accept_lang) || find_language(accept_lang.split('-').first)
+           end
+
+    lang
```

Comment:
Isn't this equivalent to purely writing the if condition? (without assignment to `lang`)